src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
+
+    static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void writeTaskToFile(String filePath) {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            for (Task taskToAdd : tasks) {
+                fw.write(taskToAdd.writeToFile() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
```

Comment:
Ideally, you shouldn't be catching errors at this step, since the fileWriter should not know what to do when an error occurs. Because you caught the error here, it actually makes the error catching code in main redundant, since it would not surface up to main.
So ideally, the file writer should tell its caller (main) that an error occurred, and it is the caller's job to handle the error.